package-res/resources/web/pentaho/type/Context.js

```diff
@@ -155,15 +240,63 @@ define([
     },
 
     /**
-     * Gets the instance constructor of a type.
+     * Gets, asynchronously, the **configured instance constructor** instance constructor of a type.
+     *
+     * For more information on the `typeRef` argument,
+     * please see [UTypeReference]{@link pentaho.type.spec.UTypeReference}.
+     *
+     * This method can be used even if a generic type metadata specification references
+     * non-standard types whose modules have not yet been loaded by the AMD module system.
+     *
+     * An error is thrown when:
```

Comment:
The proper way to add sanity checks during development without impacting production is not the BEBUG variable, You should use the 'assert' feature built right into the JVM. These checks are enabled at test time, but usually turned off during prod with no overhead penalty.

https://docs.oracle.com/javase/8/docs/technotes/guides/language/assert.html